test/rubygems/test_require.rb

```diff
@@ -471,11 +471,10 @@ def test_realworld_upgraded_default_gem
     File.write(path, code)
 
     output = Gem::Util.popen({ "GEM_HOME" => @gemhome }, *ruby_with_rubygems_in_load_path, path).strip
-    assert $?.success?
-    refute_empty output
```

Comment:
Why removing this assertion? Would provide a clear message if output is empty, no? I think moving the assertion of the status code to the end gets you the better message that you want.

Reply:
Hm :thinking: My thinking was that if the string is empty, the following assert would fail anyway. But with empty string, it would actually fail unexpectedly with `NoMethodError: undefined method 'chomp' for nil:NilClass`. This could be fixed by `output.lines(chomp: true)[0]`, which provides better error message: `<"999.99.9"> expected but was <nil>`. But all in all, you are probably right that it won't hurt to keep it around.

Reply:
Updated. Thx for the nudge :blush: